sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/TransportProducerPoolTests.cs

```diff
@@ -0,0 +1,347 @@
+        {
+            var transportProducer = new ObservableTransportProducerMock();
+            var connection = new MockConnection(() => transportProducer);
+            var retryPolicy = new EventHubProducerClientOptions().RetryOptions.ToRetryPolicy();
+            var startingPool = new ConcurrentDictionary<string, TransportProducerPool.PoolItem>
+            {
+                ["0"] = new TransportProducerPool.PoolItem("0", transportProducer)
+            };
+            TransportProducerPool transportProducerPool = new TransportProducerPool(connection, retryPolicy, startingPool);
+
+            var pooledProducer = transportProducerPool.GetPooledProducer("0", TimeSpan.FromMinutes(-1));
+
+            await using (var _ = pooledProducer.ConfigureAwait(false))
+            {
+            };
+
+            GetExpirationCallBack(transportProducerPool).Invoke(null);
+
+            Assert.That(transportProducer.CloseCallCount == 1);
+        }
+
+        /// <summary>
+        ///   The <see cref="TransportProducerPool"/> returns the right <see cref="TransportProducer" />
+        ///   matching the righ partition id.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(null)]
+        [TestCase("0")]
+        public void TransportProducerPoolAllowsTakingTheRightTransportProducer(string partitionId)
+        {
+            var transportProducer = new ObservableTransportProducerMock();
+            var partitionProducer = new ObservableTransportProducerMock(partitionId);
+            var connection = new MockConnection(() => partitionProducer);
+            var retryPolicy = new EventHubProducerClientOptions().RetryOptions.ToRetryPolicy();
+            var startingPool = new ConcurrentDictionary<string, TransportProducerPool.PoolItem>
+            {
+                ["0"] = new TransportProducerPool.PoolItem("0", partitionProducer)
+            };
+            TransportProducerPool transportProducerPool = new TransportProducerPool(connection, retryPolicy, eventHubProducer: transportProducer);
+
+            var returnedProducer = transportProducerPool.GetPooledProducer(partitionId).TransportProducer as ObservableTransportProducerMock;
+
+            Assert.That(returnedProducer.PartitionId == partitionId);
+        }
+
+        /// <summary>
+        ///   Gets the routine responsible of finding expired producers.
+        /// </summary>
+        ///
+        private static TimerCallback GetExpirationCallBack(TransportProducerPool pool) =>
+            (TimerCallback)
+                typeof(TransportProducerPool)
+                    .GetMethod("PerformExpiration", BindingFlags.NonPublic | BindingFlags.Instance)
+                    .Invoke(pool, null);
+
+        /// <summary>
+        ///   Serves as a non-functional connection for testing producer functionality.
+        /// </summary>
+        ///
+        private class MockConnection : EventHubConnection
```

Comment:
I separated those in ```CreateTransportClient```, was that the one you wanted to flag?

Reply:
My bad. I should have been more specific. I flagged the class name because there was more than one member that needed attention.

Here's the other one:
https://github.com/Azure/azure-sdk-for-net/blob/ef5ab82ac2af6f825d9f6203afc706dc51bd5737/sdk/eventhub/Azure.Messaging.EventHubs/tests/Producer/TransportProducerPoolTests.cs#L254